docs/design/ovs-pipeline.md

```diff
@@ -770,27 +784,29 @@ If you dump the flows of this table, you may see the following:
 5. table=SpoofGuard, priority=0 actions=drop
 ```
 
-Flow 1 is for case 1, matching packets received from the local Antrea gateway port without checking the source IP and MAC
-address. There are some cases where the source IP of the packets through the local Antrea gateway port is not the local
-Antrea gateway IP:
+Flow 1 is for case 1, matching packets received on the local Antrea gateway port without checking the source IP and MAC
+addresses. There are some cases where the source IP of the packets through the local Antrea gateway port is not the local
+Antrea gateway IP address:
 
 - When Antrea is deployed with kube-proxy, and `AntreaProxy` is not enabled, packets from local Pods destined for Services
-  will first go through the gateway, get load-balanced by the kube-proxy data path (DNAT) then re-enter through the
-  gateway. Then the packets are received on the gateway port with a source IP belonging to a local Pod.
+  will first go through the gateway port, get load-balanced by the kube-proxy data path (undergo DNAT with a local Endpoint
+  selected by the kube-proxy) then re-enter through the gateway port. Then the packets are received on the gateway port
```

Comment:
I may be wrong but I think that is also the case when kube-proxy selects a non-local Endpoint: I don't think kube-proxy will masquerade the traffic and the packet will go back through the gateway to be sent to the remote Pod over the appropriate tunnel. The only exception is the hairpin case.

So maybe replace "undergo DNAT with a local Endpoint selected by the kube-proxy" with "undergoes DNAT", unless I am missing something?

Reply:
You reminded me. When kube-proxy selects a non-local Endpoint, after DNAT, the packets will be forwarded to antrea-gw0 by the onlink routing added by Antrea NodeController.

Reply:
Added something to remind the reader like this:

When Antrea is deployed with kube-proxy, and `AntreaProxy` is not enabled, packets from local Pods destined for Services
  will first go through the gateway port, get load-balanced by the kube-proxy data path (undergoes DNAT) then re-enter
  the OVS pipeline through the gateway port **(through an "onlink" route directing Pod CIDR traffic to the gateway port)**